databricks-sdk-java/src/main/java/com/databricks/sdk/core/DatabricksConfig.java

```diff
@@ -209,6 +212,23 @@ public synchronized Map<String, String> authenticate() throws DatabricksExceptio
     }
   }
 
+  public TokenSource getTokenSource() {
+    try {
+      if (headerFactory == null) {
+        ConfigLoader.fixHostIfNeeded(this);
+        headerFactory = credentialsProvider.configure(this);
+        setAuthType(credentialsProvider.authType());
+      }
+      if (headerFactory instanceof OAuthHeaderFactory) {
+        return (TokenSource) headerFactory;
+      }
+      return new ErrorTokenSource(
+          String.format("OAuth Token not supported for current auth type %s", authType));
+    } catch (Exception e) {
+      return new ErrorTokenSource("Failed to get token source: " + e.getMessage());
+    }
```

Comment:
Restrict the scope of the try catch to the specific instruction that might throw an exception. IIUC, this would be `configure`?

```suggestion
      if (headerFactory == null) {
        try {
          ConfigLoader.fixHostIfNeeded(this);
          headerFactory = credentialsProvider.configure(this);
        } catch (Exception e) {
          return new ErrorTokenSource("Failed to get token source: " + e.getMessage());
        }     
        setAuthType(credentialsProvider.authType());
      }
      
      if (headerFactory instanceof OAuthHeaderFactory) {
        return (TokenSource) headerFactory;
      }
      return new ErrorTokenSource(
          String.format("OAuth Token not supported for current auth type %s", authType));
```

Reply:
FIxed!